pkg/api/types.go

```diff
@@ -44,6 +44,7 @@ type Properties struct {
 	AgentPoolProfiles       []AgentPoolProfile      `json:"agentPoolProfiles"`
 	LinuxProfile            LinuxProfile            `json:"linuxProfile"`
 	WindowsProfile          WindowsProfile          `json:"windowsProfile"`
+	ExtensionsProfile       []ExtensionProfile      `json:"extensionsProfile"`
```

Comment:
Also, I have created another branch and have DC/OS support completed.  After this PR is complete, I will issue another PR with DC/OS